lib/http2.c

```diff
@@ -1820,6 +1820,9 @@ static ssize_t stream_recv(struct Curl_cfilter *cf, struct Curl_easy *data,
   struct stream_ctx *stream = H2_STREAM_CTX(data);
   ssize_t nread = -1;
 
+  if(!stream)
+    return nread;
```

Comment:
```suggestion
  if(!stream) {
    *err = CURLE_AGAIN;
    return nread;
  }
```

I'd see what @icing has to say about it, there may be a bigger issue here that the unity build caught that shouldn't exist in the first place.

Reply:
Absolutely. We also might want enable unity on a H2/H3 build (or add one) in the CI to catch these early.